hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java

```diff
@@ -383,36 +383,29 @@ public void deleteByControllerID(final String controllerID) {
 
     @Override
     public Page<Target> findByAssignedDistributionSet(final Pageable pageReq, final long distributionSetID) {
-        throwEntityNotFoundIfDsDoesNotExist(distributionSetID);
+        final DistributionSet validDistSet = distributionSetManagement.getValidAndComplete(distributionSetID);
```

Comment:
please verify each method if it really makes sense to only get valid Distribution sets, or should we just verify existence with `distributionSetManagement#getOrElseThrowException`. E.g. in this case `findByAssignedDistributionSet` it is used for getting the assigned targets for particular DS via Mgmt Api and during "pinning" of DS in Deployment View. If my assumption is correct, invalidated DS that is still shown on Deployment View will not support "pinning" anymore, but rather show an error